src/batch/src/executor/row_seq_scan.rs

```diff
@@ -319,28 +318,28 @@ impl<S: StateStore> RowSeqScanExecutor<S> {
         }
 
         // Range Scan
-        let range_scans = select_all(range_scans.into_iter().map(|range_scan| {
-            let table = table.clone();
-            let histogram = histogram.clone();
-            Box::pin(Self::execute_range(
-                table,
-                range_scan,
+        // WARN: DO NOT use `select` to execute range scans concurrently
+        //       it can consume too much memory if there're too many ranges.
+        for range in range_scans {
+            let stream = Self::execute_range(
```

Comment:
Not sure if it should be a default. I'm thinking a session variable can easily enable / disable this feature.

Reply:
Perhaps we can do something like:
if range_scans.len() > some_arbitrary_threshold, 
then do sequential. Else do parallel.

Reply:
Make it default LGTM. Previously, point-lookup is also performed concurrently, but I changed it to sequential, because lookup-join encountered a similar issue as well.